src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor{TPixel}.cs

```diff
@@ -187,6 +188,8 @@ private static PixelConversionModifiers GetModifiers(bool compand, bool premulti
             bool compand,
             bool premultiplyAlpha)
         {
+            bool pixelHasNoAlpha = PixelOperations<TPixel>.Instance.GetPixelTypeInfo()?.AlphaRepresentation == PixelAlphaRepresentation.None;
+            premultiplyAlpha &= !pixelHasNoAlpha;
```

Comment:
@JimBobSquarePants I couldn't figure out how should this *really* work with our existing pixel types, so went with the safest option here (only omit premultiplication when we are 100% sure that the alpha channel is missing).

However, if I understand it correctly normally we should only premultiply in case of `PixelAlphaRepresentation.Associated`. What confuses me is that `Rgba32` and friends are configured to `Unassociated`:

https://github.com/SixLabors/ImageSharp/blob/97dde7fb2f0ac97b5b1f9e6b0555a53d2e270c46/src/ImageSharp/PixelFormats/PixelImplementations/PixelOperations/Rgba32.PixelOperations.cs#L22-L23

Reply:
Ah no. Associated alpha means premultiplied. The only occasion we should premultiply is `PixelAlphaRepresentation.Unassociated`
https://en.wikipedia.org/wiki/Alpha_compositing#Straight_versus_premultiplied

Reply:
Thanks, I see now!

Would be the following code better here then?
```suggestion
            PixelAlphaRepresentation? alphaRepresentation = PixelOperations<TPixel>.Instance.GetPixelTypeInfo()?.PixelAlphaRepresentation;

            // Premultiply only if alpha representation is unknown or Unassociated:
            bool needsPremultiplication = alphaRepresentation == null || alphaRepresentation.Value == PixelAlphaRepresentation.Unassociated;
            premultiplyAlpha &= needsPremultiplication;
```

Reply:
Yep. That looks good to me. 👍